### PR TITLE
renames Symbol.kind and Type.kind in the API

### PR DESCRIPTION
**Pull request by xeno-by**
Kind as in http://en.wikipedia.org/wiki/Kind_(type_theory) seems to be
too useful of a name to bind it to a debugging facility in the API.

I also added a comment that this facility is meant for debugging only
and should not be used to discriminate symbols/types in production code.

### CODE REVIEW

**Comment by xeno-by** [comment]
review by @paulp, @retronym

**Comment by xeno-by** [comment]
PLS REBUILD ALL

File: src/reflect/scala/reflect/api/Symbols.scala [hunk 1]
```diff
@@ -201,8 +201,15 @@ trait Symbols extends base.Symbols { self: Universe =>
...
-    def kind: String
+    /** The string discriminator of this symbol.
+     *
+     *  Useful for debugging and debug printing.
+     *
+     *  Values returned by this method should not be used to tell symbols apart.
+     *  These values can changed on a whim, depending on how the underlying implementation sees fit.
+     *  Use one of the `isXXX` methods instead (e.g. isMethod or isGetter).
+     */
+    def discr: String
```
Comment (soc) [inline src/reflect/scala/reflect/api/Symbols.scala:212]: What's wrong with something like `def debugString`?

Comment (xeno-by) [inline src/reflect/scala/reflect/api/Symbols.scala:212]: `debugString` is a nice name, but inside the compiler it's used for something else: https://github.com/scalamacros/kepler/blob/fccee33223468f2887951e8c27606106d37363a6/src/reflect/scala/reflect/internal/TypeDebugging.scala#L70

**Comment by paulp** [comment]
Discr? Absolutely not.

**Comment by xeno-by** [comment]
Any suggestions?

**Comment by xeno-by** [comment]
Also why close pull request? The renaming is unnecessary? The name is bad? Anything else?

**Comment by paulp** [comment]
I'll give some suggestions when I'm not on a phone; the closing was to be emphatic.

**Pull request closed without merging**